### ISSUE TITLE
Typical MS DOCs: GetJsonAsync

### ISSUE THREAD

**Issue by BattlefieldHou** [aspnet/AspNetCore.Docs#15843]
This error in the example has still not been address.  I have always found Microsoft Docs to be the last resort after StackOverflow has been totally searched and then come here to find nothing ever makes sense or works.  The partial examples and complete lack of flow on how to get examples to work is just so frustrating on almost everything linked to Microsoft Docs.  Sure hope they don't buy StackOverflow!

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8b5e2e10-ff7b-2b9f-84da-f16028ae2c53
* Version Independent ID: c7e59a08-1c60-32c2-75fd-33cb77ff7a5d
* Content: [Call a web API from ASP.NET Core Blazor](https://docs.microsoft.com/en-us/aspnet/core/blazor/call-web-api?view=aspnetcore-3.0#feedback)
* Content Source: [aspnetcore/blazor/call-web-api.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/call-web-api.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
See: https://github.com/aspnet/AspNetCore.Docs/issues/15430#issuecomment-553644665

Since a few developers are hitting this, I'll note it in the topic.

**Comment by guardrex** [comment]
The topic tries to explain that it's focused on **_Blazor WebAssembly_** and the experimental package. The second line says ...

> Blazor Server apps call web APIs using `HttpClient` instances typically created using `IHttpClientFactory`. For more information, see [Make HTTP requests using IHttpClientFactory in ASP.NET Core](https://docs.microsoft.com/aspnet/core/fundamentals/http-requests).

... and that link takes the reader over to the topic that they should follow (indeed, where the reader **_won't see_** `GetJsonAsync`).

Now that a couple of these remarks have come in (I think you're the second to ask), I take it that the first two lines introducing the topic and the following call outs ...

* "For Blazor WebAssembly ..."
* "In Blazor WebAssembly ..."
* "When running on WebAssembly in a Blazor WebAssembly app, ..."

... don't go far enough that the `HttpClient` for Blazor Server isn't the service (and examples) described by this topic. On a broader point, I'm a little concerned about the **_two hosting models_** being sufficiently clear in readers minds to pick up on these distinctions because many topics use similar language to call out concepts that pertain to one hosting model or the other. I'll keep an :ear: open for that happening as we go.

For this issue, I'll clarify it further so that readers don't get tripped up. Thanks for writing in. I'll ping you on the PR.

**Comment by BattlefieldHou** [comment]
I think MS docs makes a huge assumption of knowledge when looking at the docs. Even you pointing out the paragraph that should how have redirected me to another page, I’m still not sure why that tells me to leave that example alone and go to another page. I guess I only know enough to be dangerous (and frustrated). Thanks for listening to our feedback though. 

My whole objective was to figure out how to utilize a web api call inside of blazor. I’m ok with server side but couldn’t put that puzzle together from any of the docs. That’s probably too much of a newbie thing to search but I’m still not able to do it.

**Comment by guardrex** [comment]
Yes, I thought that was the case ... it's *earlier* in the coverage.

One has to recognize that Blazor Server and Blazor WebAssembly are **_two different hosting models_** and thus sometimes how to do things will diverge ... in this case, the coverage diverges a lot. In this case, this entire topic is for Blazor WebAssembly. With Blazor WebAssembly, everything here works. When using the Blazor Server hosting model, **_nothing here works_**.

It's covered earlier ... https://docs.microsoft.com/en-us/aspnet/core/blazor/hosting-models?view=aspnetcore-3.0

If you haven't already seen it on the PR, I'm going a lot further to try and get readers to the right spot. In the end tho, they will need to know from reading the earlier topic what the difference is.

... indeed, the best way to learn Blazor via the docs from stratch is to read the topics from the top of that TOC (table of contents) node right down the list.

**Comment by BattlefieldHou** [comment]
Thanks. I had read them and watched a ton of videos but didn’t realize I was in webassembly until I downloaded the sample code. I was hoping I could somehow still make it work but as you pointed out, that wasn’t going to happen. I found that page by searching how to do web api through blazor and unfortunately I didn’t get the context of the page as it was standalone. I think that’s typical of MS docs that the reader would be reading from the beginning and not jump directly to that topic or in the middle of it due to a search. 

Again, thanks. Stop working on your day off!  I’m old and can assure you that you don’t get that time back. :)
